tests/SprykerTest/Shared/AppPayment/_support/Helper/AppPaymentHelper.php

```diff
@@ -149,11 +149,7 @@ public function haveInitializePaymentRequestWithAppConfigTransfer(array $seed =
 
         $quoteTransfer = $quoteBuilder->build();
 
-        if ($additionalPaymentData) {
-            $paymentTransfer = $quoteTransfer->getPayment() ?? new PaymentTransfer();
-            $paymentTransfer->setAdditionalPaymentData($additionalPaymentData);
-            $quoteTransfer->setPayment($paymentTransfer);
-        }
+        $quoteTransfer->setAdditionalPaymentData($additionalPaymentData);
```

Comment:
Why don't you use Payment.additionalPaymentData? this data related to payment

Reply:
on SCOS the structure is `QuoteTransfer.payment.additionalPaymentData` but it was always sent as `orderData.additionalPaymentData` (flat structure), so `orderData.payment` never existed.